### ISSUE TITLE
Using ProxyServer

### ISSUE THREAD

**Issue by ovhelemann** [neopsis/niagara-weather#2]
Hi,

I encountered the problem, that this module does not work with Niagara Proxy. I have to use it here with httpProxyService from net-rt module. 
In your code you use HttpURLConnection, but if the stream should go through Niagara ProxyService it has to be used javax.baja.net.HttpConnection. Is it elaborately to change this?

Thanks in advance!
OVH

**Comment by ovhelemann** [comment]
I tried to fix it myself and it seems to work fine.

I added net-rt module to gradle file and replaced HttpUrlConnection with HttpConnection Object

`HttpConnection conn = (HttpConnection) new HttpConnection(new BIpHost("api.openweathermap.org"), 80, link);
conn.setTimeout((int) BRelTime.makeSeconds(30).getMillis());
conn.connect();`

There were some other parts that I changed. I could send you my files if needed.

**Comment by envas** [comment]
@ovhelemann - thank you, I modified the code in the release 4.0.2. 

However, I did not apply your connection response handling. In your code, when the server responses with the status <> 200, the InputStream becomes null. This, in turn, fails in the next row with NullPointerException when reading the response. 

Openweathermap, in case of a syntax error, returns the valid JSON response with the error description in the `message` JSON element. The HTTP status is <> 200. Here is an example of the response for request with an invalid location. Returned HTTP status is 404. 

![Openweathermap](https://user-images.githubusercontent.com/25388570/72171651-b40ba700-33d3-11ea-811e-f87b3a40428d.png)

When the API key is invalid, the return status is 401, etc. It is important to propagate the return message to the Niagara property `Fault Cause` to get the user more detailed info.
